src/buffer.cpp

```diff
@@ -38,5 +38,11 @@ const void* Buffer::data() const {
 size_t Buffer::getByteSize() const {
     return byteSize;
 }
+const BufferType Buffer::getBufferType() const {
+    return bufferType;
+}
+const std::optional<uint32_t> Buffer::getDeviceId() const {
```

Comment:
return reference instead of copy?

Reply:
getBufferType can remain with copy since it's basic type but here we should have ref